qa/L0_mlflow/plugin_test.py

```diff
@@ -114,6 +115,30 @@ def test_onnx_flavor_with_files(self):
             filecmp.cmp(config_path,
                         "./models/onnx_model_with_files/config.pbtxt"))
 
+    @mock_s3
+    def test_s3_support(self):
+        # Set TRITON_MODEL_REPO to an s3 path.
+        os.environ["TRITON_MODEL_REPO"] = "s3://moto-bucket/"
+
+        # Set up moto's AWS
+        import boto3
+        client = boto3.resource("s3", region_name="eu-central-1")
+        client.create_bucket(Bucket="moto-bucket")
+        del client
+
+        # Log the ONNX model to MLFlow
+        import mlflow.onnx
+        import onnx
+        model = onnx.load(
+            "./mlflow-triton-plugin/examples/onnx_float32_int32_int32/1/model.onnx"
+        )
+
+        # Use a different name to ensure the plugin operates on correct model
+        mlflow.onnx.log_model(model,
+                              "triton",
+                              registered_model_name="onnx_model")
+
+        self._validate_deployment("onnx_model")
```

Comment:
I am looking into making this an valid test. The issue is that `TRITON_MODEL_REPO` should also be the path that Triton poll the model from, which should be a valid S3 path. The mock here only overwrite the client side of things and the server will try to poll from the path expecting it to be a valid S3 path. I think the `test.sh` need to be modified to properly setup a S3 bucket and the `plugin_test.py` shouldn't need to change, running the existing tests (`test_onnx_xxx`) should be sufficient as the plugin should internally detect `TRITON_MODEL_REPO` to be a S3 location and upload the files accordingly

Reply:
I still think we need to mock the s3 connection, otherwise the boto3 client will try to connect to an s3 bucket that is not mocked nor exists. Maybe we just need a pre-action for the s3 test that uploads the model to the mocked s3 environment so the model can be pulled from TRITON_MODEL_REPO. What do you think?

Reply:
Note that Triton server is running along side with the plugin, so Triton server will try to poll `TRITON_MODEL_REPO` which is not an actual bucket on S3 (it is actually local), so Triton can't properly poll the model files and load the model, unless the same mock is injected in Triton server.

Also, the added test is not relaunching the Triton server to monitor the S3 model repository, so it appears to pass because the previous tests have coincidentally setup local model repository with the same model name `onnx_model` and passed the validation.

I have updated the test to set up Triton for S3 separately in this commit https://github.com/triton-inference-server/server/commit/86fb33ca050173680c7f73c8fca0d9daa253e8bd, do you mind to adopt that and test the plugin changes? I ran it locally and it seems like the test failed because the plugin only uploaded part of the model files (only `mlflow-meta.json` and the actual `model.onnx` is missing):
```
# server log
E0411 19:01:34.283724 11677 model_repository_manager.cc:1245] Poll failed for model directory 'onnx_model': Invalid model name: Could not determine backend for model 'onnx_model' with no backend in model configuration. Expected model name of the form 'model.<backend_name>'.

# bucket content
aws s3 rm s3://triton-bucket-gluo-mlflow/ --recursive --include '*'
delete: s3://triton-bucket-gluo-mlflow/onnx_model_with_files/mlflow-meta.json
delete: s3://triton-bucket-gluo-mlflow/onnx_model/mlflow-meta.json
```

Reply:
To run the test local, you need to set the s3 environment variables `AWS_DEFAULT_REGION`, `AWS_ACCESS_KEY_ID` `AWS_SECRET_ACCESS_KEY`, and provide `CI_JOB_ID` to construct the bucket name used in the test before running the bash script.

Reply:
> Note that Triton server is running along side with the plugin, so Triton server will try to poll TRITON_MODEL_REPO which is not an actual bucket on S3 (it is actually local), so Triton can't properly poll the model files and load the model, unless the same mock is injected in Triton server.

That's right, I didn't think about it. I'll include the commit you shared (btw, thanks!) and will try to see why it fails. I haven't been able to run the CI yet, still facing several issues, but maybe I can try to run the test bash script if I can pull the needed resources to run the test. I'll let you know ASAP!

Reply:
I have managed to set up a local QA environment. I think I am getting more or less where the issue comes from: I tested this with a Tensorflow model and `triton` flavor, which assumes the model is already in the format/structure that Triton expects, that is: 
```
model
├── 1
│   └── model.savedmodel
│       ├── assets
│       ├── keras_metadata.pb
│       ├── saved_model.pb
│       └── variables
│           ├── variables.data-00000-of-00001
│           └── variables.index
└── config.pbtxt
```

The issue with onnx is that it works in a different way which I didn't test. I'm working on it now so it can get solved ASAP! 

Thanks so much for your additions to `test.sh`, it made it really easy to test this changes! I'll keep you posted.